File: api/src/main/java/org/pmiops/workbench/actionaudit/ActionAuditService.java
package org.pmiops.workbench.actionaudit;

import java.util.Collection;
import java.util.Set;

public interface ActionAuditService {
  default void send(ActionAuditEvent event) {
    send(Set.of(event));
  }

  void send(Collection<ActionAuditEvent> events);
}

File: api/src/main/java/org/pmiops/workbench/actionaudit/ActionAuditService.kt
package org.pmiops.workbench.actionaudit

import java.util.logging.Logger

interface ActionAuditService {
    fun send(event: ActionAuditEvent) {
        send(setOf(event))
    }

    fun send(events: Collection<ActionAuditEvent>)

    fun logRuntimeException(logger: Logger, exception: RuntimeException)
}
